extensions/format/format.lisp

```diff
@@ -0,0 +1,41 @@
+(defpackage #:lem-format
+  (:use :cl :lem)
+  (:export #:auto-format? 
+           #:register-formatter 
+           #:register-formatters 
+           #:format-buffer))
+(in-package :lem-format)
+
+;; Set this to true to format on save.
+(defvar auto-format? nil)
```

Comment:
style: `*auto-format*` ? Lem doesn't have a style guide yet, but generally we use `*earmuffs*`. It's the first time I see "?" in the code base too, also lately cxxxr told me he preferred to _not_ use a suffix (I was using "-p") for a user-level parameter.

Reply:
Not sure if I like removing the suffix, but I will do it to keep in line with cxxxr's style.